### DISCUSSION TITLE
[v4 websocket] Is there a way to get Socket ID on the frontend?

### DISCUSSION THREAD

**Discussion by supercede** [adonisjs/core#2377] [Help]
Using socket io, you can get the ID for a particular socket. Is there a way to do something similar on adonis from the client side using adonis-websocket-client with the adonis websocket library? From the API, I know it's socket.id. Just wondering if the value can be got on the client

**Comment by McSneaky** [comment]
You can send it to the client over WS as a first message 😉

**Reply by supercede** [reply]
I was going to try that but I found this more suited to my needs:

```
ws.on('open', (socket) => {
      socketId = `${topic}#${socket.connId}`
}
